### PR TITLE
Psums sol

### PR DESCRIPTION
**Pull request by HarryWangATX**

### CODE REVIEW

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 1]
```diff
@@ -439,6 +439,86 @@ int main() {
...
+            int x1 = io.nextInt();
+            int y1 = io.nextInt();
+            int x2 = io.nextInt();
+            int y2 = io.nextInt();
+            io.println(pfx[x2][y2] - pfx[x1-1][y2] - pfx[x2][y1-1] + pfx[x1-1][y1-1]);
+        }
+        io.close();
+    }
+
+    private static class Kattio extends PrintWriter {
```
Comment (dongliu0426) [inline content/3_Silver/Prefix_Sums_2.mdx:487]: replace this with `CodeSnip{Kattio}`?

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
tabs :eyes:

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 2]
```diff
@@ -439,6 +439,56 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class ForestQueries{
```
Comment (dongliu0426) [inline content/3_Silver/Prefix_Sums_2.mdx:448]:
```suggestion
public class ForestQueries {
```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 3]
```diff
@@ -439,6 +439,56 @@ int main() {
...
+	static int N;
+	static int Q;
+	static int[][] pfx;
+	static int[][] arr;
+	public static void main(String[] args){
```
Comment (dongliu0426) [inline content/3_Silver/Prefix_Sums_2.mdx:453]:
```suggestion
	public static void main(String[] args) {
```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 4]
```diff
@@ -439,6 +439,56 @@ int main() {
...
+		N = io.nextInt();
+		Q = io.nextInt();
+
+		pfx = new int[N+1][N+1];
+		arr = new int[N+1][N+1];
+
+		for(int i = 0; i < N; i++) {
+			String line = io.next();
+			for(int j = 0; j < N; j++) {
+				if(line.charAt(j) == '*'){
```
Comment (dongliu0426) [inline content/3_Silver/Prefix_Sums_2.mdx:465]:
```suggestion
				if(line.charAt(j) == '*') {
```

**Review by dongliu0426 — APPROVED** [review]
spacing stuff

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 5]
```diff
@@ -439,6 +439,56 @@ int main() {
...
+		for(int i = 0; i < Q; i++) {
+			int x1 = io.nextInt();
+			int y1 = io.nextInt();
+			int x2 = io.nextInt();
+			int y2 = io.nextInt();
+			io.println(pfx[x2][y2] - pfx[x1-1][y2] - pfx[x2][y1-1] + pfx[x1-1][y1-1]);
+		}
+		io.close();
+	}
+	CodeSnip{Kattio}
```
Comment (frodakcin) [inline content/3_Silver/Prefix_Sums_2.mdx:486]:
```suggestion

	CodeSnip{Kattio}
```

Looks nicer with extra space.

Also, this doesn't compile... Kattio needs to be declared as "static"

Comment (frodakcin) [reply content/3_Silver/Prefix_Sums_2.mdx:486]: However that will be a tomorrow problem (unless sm1 else takes care of it) cuz I gon go to sleep

**Review by frodakcin — APPROVED** [review]

**Pull request merged**